### PR TITLE
Split version number with regex

### PR DESCRIPTION
**Pull request by yol**
Third time's the charm!

### CODE REVIEW

File: indexer/indexer.py [hunk 1]
```diff
@@ -38,9 +38,9 @@ def split_version(path):
     # e.g. script.module.idna-2.8.zip, script.module.idna-2.8.1.zip, ..., script.module.idna-2.8.1.2.3.4.zip
     #      script.module.idna-2.8.10+git010cab3.zip script.module.idna-2.8.10+matrix.2.zip
     #      resource.images.languageflags-flat-0.0.1.zip resource.language.sr_rs@latin-3.0.9.zip
-    parts = result[0].rsplit("-", 1)
-    if parts and re.match(r"^[\w.\-@]+$", parts[0]) and re.match(r"^\d+\.\d+(\.\d+){0,4}([+~\w.]+)?$", parts[1]):
-        return tuple(parts)
+    m = re.match(r"^([\w\.]+.*)-(\d+\.\d+(\.\d+){0,4}([+~\w]+(\.\d+)?)?)$", result[0])
```
Comment (yol) [inline indexer/indexer.py:41]: Should we be more strict with package names and not allow `.*`?

**Comment by wsnipex** [comment]
I wouldn't. We've tried the sticter approaches and failed :P

**Comment by yol** [comment]
The regex that is there now (`[\w.\-@]+`) did match all the packages

**Comment by yol** [comment]
I guess it's fine if just the add-on checker uses the stricter variant @enen92

**Comment by enen92** [comment]
Maybe it's better to add a check for the version instead of the regex, that way we have a bit more freedom to check whatever we see fit. I'll try to have a look after work

**Pull request merged**